### ISSUE TITLE
Fix warnings in net unit tests

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#5709]
Run `./mach test-unit` to see those:

```
/tests/unit/net/mime_classifier.rs:5:5: 5:22 warning: use of unstable library feature 'old_io': replaced with std::fs::File
/tests/unit/net/mime_classifier.rs:5 use std::old_io::File;
                                         ^~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:5:22: 5:22 help: add #![feature(old_io)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:6:5: 6:12 warning: use of unstable library feature 'os'
/tests/unit/net/mime_classifier.rs:6 use std::os;
                                         ^~~~~~~
/tests/unit/net/mime_classifier.rs:6:12: 6:12 help: add #![feature(os)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:15:13: 15:22 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:15     let p = Path::new("../../tests/unit/net/parsable_mime/video/mp4/test.mp4");
                                                  ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:15:22: 15:22 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:16:20: 16:30 warning: use of unstable library feature 'old_io': replaced with std::fs::File::open
/tests/unit/net/mime_classifier.rs:16     let mut file = File::open(&p);
                                                         ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:16:30: 16:30 help: add #![feature(old_io)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:17:28: 17:41 warning: use of unstable library feature 'old_io'
/tests/unit/net/mime_classifier.rs:17     let read_result = file.read_to_end();
                                                                 ^~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:17:41: 17:41 help: add #![feature(old_io)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:30:36: 30:40 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:30 fn test_sniff_full(filename_orig: &Path,type_string: &str,subtype_string: &str,
                                                                         ^~~~
/tests/unit/net/mime_classifier.rs:30:40: 30:40 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:32:37: 32:47 warning: use of unstable library feature 'os'
/tests/unit/net/mime_classifier.rs:32     let current_working_directory = os::getcwd().unwrap();
                                                                          ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:32:47: 32:47 help: add #![feature(os)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:33:71: 33:80 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:33     println!("The current directory is {}", current_working_directory.display());
                                                                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:33:80: 33:80 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:35:24: 35:33 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:35     let mut filename = Path::new("../../tests/unit/net/parsable_mime/");
                                                             ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:35:33: 35:33 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:37:14: 37:33 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:37     filename.push(filename_orig);
                                                   ^~~~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:37:33: 37:33 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:40:20: 40:30 warning: use of unstable library feature 'old_io': replaced with std::fs::File::open
/tests/unit/net/mime_classifier.rs:40     let mut file = File::open(&filename);
                                                         ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:40:30: 40:30 help: add #![feature(old_io)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:41:28: 41:41 warning: use of unstable library feature 'old_io'
/tests/unit/net/mime_classifier.rs:41     let read_result = file.read_to_end();
                                                                 ^~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:41:41: 41:41 help: add #![feature(old_io)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:46:37: 46:47 warning: use of unstable library feature 'core': Instead of taking this bound generically, this trait will be replaced with one of slicing syntax (&foo[..]), deref coercions, or a more generic conversion trait
/tests/unit/net/mime_classifier.rs:46                     if (parsed_type.as_slice() != type_string) ||
                                                                          ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:46:47: 46:47 help: add #![feature(core)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:47:39: 47:49 warning: use of unstable library feature 'core': Instead of taking this bound generically, this trait will be replaced with one of slicing syntax (&foo[..]), deref coercions, or a more generic conversion trait
/tests/unit/net/mime_classifier.rs:47                         (parsed_subtp.as_slice() != subtype_string) {
                                                                            ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:47:49: 47:49 help: add #![feature(core)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:49:45: 49:53 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:49                                    filename.as_str().unwrap(), type_string, subtype_string,
                                                                                  ^~~~~~~~
/tests/unit/net/mime_classifier.rs:49:53: 49:53 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:54:41: 54:49 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:54                                filename.as_str().unwrap(), supplied_type),
                                                                              ^~~~~~~~
/tests/unit/net/mime_classifier.rs:54:49: 54:49 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:58:35: 58:43 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:58                          filename.as_str().unwrap(), e),
                                                                        ^~~~~~~~
/tests/unit/net/mime_classifier.rs:58:43: 58:43 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:65:17: 65:26 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:65     let mut x = Path::new("./");
                                                      ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:65:26: 65:26 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:66:7: 66:24 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:66     x.push(type_string);
                                            ^~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:66:24: 66:24 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:67:7: 67:27 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:67     x.push(subtype_string);
                                            ^~~~~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:67:27: 67:27 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:68:7: 68:17 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:68     x.push(file);
                                            ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:68:17: 68:17 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:173:22: 173:31 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:173     test_sniff_full(&Path::new("unknown/true_type.ttf"), "(TrueType)", "", None);
                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:173:31: 173:31 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:179:22: 179:31 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:179     test_sniff_full(&Path::new("unknown/open_type"), "(OpenType)", "", None);
                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:179:31: 179:31 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:185:22: 185:31 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:185     test_sniff_full(&Path::new("unknown/true_type_collection.ttc"), "(TrueType Collection)", "", None);
                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:185:31: 185:31 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:427:22: 427:31 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:427     test_sniff_full(&Path::new("text/xml/feed.rss"), "application", "rss+xml", Some(("text", "html")));
                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:427:31: 427:31 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/mime_classifier.rs:432:22: 432:31 warning: use of unstable library feature 'old_path'
/tests/unit/net/mime_classifier.rs:432     test_sniff_full(&Path::new("text/xml/feed.atom"), "application", "atom+xml", Some(("text", "html")));
                                                            ^~~~~~~~~
/tests/unit/net/mime_classifier.rs:432:31: 432:31 help: add #![feature(old_path)] to the crate attributes to silence this warning
/tests/unit/net/image_cache_task.rs:286:21: 286:58 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:286                     chan.send(Payload(test_image_bin()));
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:287:21: 287:45 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:287                     chan.send(Done(Ok(())));
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:288:21: 288:50 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:288                     image_bin_sent_chan.send(());
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:291:21: 291:56 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:291                     resource_task_exited_chan.send(());
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:312:5: 312:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:312     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:335:21: 335:58 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:335                     chan.send(Payload(test_image_bin()));
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:336:21: 336:58 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:336                     chan.send(Done(Err("".to_string())));
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:337:21: 337:50 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:337                     image_bin_sent_chan.send(());
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:340:21: 340:56 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:340                     resource_task_exited_chan.send(());
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:363:5: 363:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:363     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:400:5: 400:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:400     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:436:5: 436:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:436     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:466:5: 466:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:466     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:494:5: 494:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:494     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:514:5: 514:24 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:514     wait_chan.send(());
                                            ^~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:522:5: 522:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:522     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:542:5: 542:24 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:542     wait_chan.send(());
                                            ^~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:550:5: 550:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:550     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/image_cache_task.rs:573:5: 573:47 warning: unused result which must be used, #[warn(unused_must_use)] on by default
/tests/unit/net/image_cache_task.rs:573     mock_resource_task.send(ControlMsg::Exit);
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:5:5: 5:22 warning: use of deprecated item: replaced with std::fs::File, #[warn(deprecated)] on by default
/tests/unit/net/mime_classifier.rs:5 use std::old_io::File;
                                         ^~~~~~~~~~~~~~~~~
/tests/unit/net/mime_classifier.rs:16:20: 16:30 warning: use of deprecated item: replaced with std::fs::File::open, #[warn(deprecated)] on by default
/tests/unit/net/mime_classifier.rs:16     let mut file = File::open(&p);
                                                         ^~~~~~~~~~
/tests/unit/net/mime_classifier.rs:40:20: 40:30 warning: use of deprecated item: replaced with std::fs::File::open, #[warn(deprecated)] on by default
/tests/unit/net/mime_classifier.rs:40     let mut file = File::open(&filename);
                                                         ^~~~~~~~~~
```

**Comment by marcusklaas** [comment]
I'd like to have a stab at this.

**Comment by jdm** [comment]
Go for it!

**Comment by neojski** [comment]
Fixed via #5734, @jdm?
